tests/lib/devicetree/api/src/main.c

```diff
@@ -1564,6 +1564,17 @@ static void test_clocks(void)
 	zassert_true(DT_SAME_NODE(DT_CLOCKS_CTLR_BY_NAME(TEST_TEMP, clk_b),
 				  DT_NODELABEL(test_clk)), "");
 
+	/* DT_NUM_CLOCKS */
+	zassert_equal(DT_NUM_CLOCKS(TEST_TEMP), 3, "");
```

Comment:
I'd like to see coverage where the number of clocks is 0 added here and for the INST version, please.

Reply:
For now, it is not possible to call this macro on a node with no "clocks" property.
I could possibly implement `DT_NUM_CLOCKS` using `DT_PROP_LEN_OR` which would return 0 is no clocks are provided, but I'm not sure this is really valid (empty clocks property vs no clocks property).
I could also mention this in macro docstring but this is likely valid for a lot of macros on other properties which don't have a fallback. So I'm not really sure this is required.

So is that really required ?

Reply:
> So is that really required ?

No, that's fine, let's ignore this request from me then. Thanks for justification